packages/arb-avm-cpp/data_storage/src/arbcore.cpp

```diff
@@ -989,8 +1028,7 @@ rocksdb::Status ArbCore::reorgCheckpoints(
 
         auto reorg_machine_begin_time = std::chrono::steady_clock::now();
         auto nearest_machine_or_status = reorgToFirstMatchingMachineCheckpoint(
-            selected_machine_output, check_output, tx, checkpoint_it,
-            value_cache);
+            check_output, tx, checkpoint_it, value_cache);
```

Comment:
I think the handling of `selected_machine_output` is off. Currently it's set to the newest valid checkpoint whereas I think it should be the `max(current val, nearest_machine)`

Reply:
refactored